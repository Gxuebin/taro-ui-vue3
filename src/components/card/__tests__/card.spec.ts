import { mount } from '@vue/test-utils'
import AtCard from '../index'

const factory = (
  values = {},
  slots = { default: ['这也是内容区 可以随意定义功能'] }
) => {
  return mount(AtCard as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values, title: '这是个标题' },
  })
}

describe('Card', () => {
  it('should render default Card', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- thumb', () => {
    const wrapper = factory({
      thumb:
        'http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- note', () => {
    const wrapper = factory({
      note: '小Tips',
      thumb:
        'http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- extra', () => {
    const wrapper = factory({
      note: '小Tips',
      extra: '额外信息',
      thumb:
        'http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- isFull', () => {
    const wrapper = factory({
      isFull: true,
      note: '小Tips',
      extra: '额外信息',
      thumb:
        'http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- extraStyle', () => {
    const wrapper = factory({
      isFull: true,
      note: '小Tips',
      extra: '额外信息',
      extraStyle: { fontSize: '12px', maxWidth: '200px' },
      thumb:
        'http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Card Behavior ', () => {
  it('should trigger onClick', async () => {
    const onClick = jest.fn()

    const wrapper = factory({
      onClick: onClick,
    })

    await wrapper.find('.at-card').trigger('tap')
    expect(onClick).toBeCalled()
  })
})
