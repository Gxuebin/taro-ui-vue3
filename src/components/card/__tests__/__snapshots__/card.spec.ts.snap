// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render default Card 1`] = `
<view
  class="at-card"
>
  <view
    class="at-card__header"
  >
    
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    
  </view>
</view>
`;

exports[`Card should render prop -- extra 1`] = `
<view
  class="at-card"
>
  <view
    class="at-card__header"
  >
    <view
      class="at-card__header-thumb"
    >
      <image
        class="at-card__header-thumb-info"
        mode="scaleToFill"
        src="http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png"
      />
    </view>
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    <text
      class="at-card__header-extra"
    >
      额外信息
    </text>
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    <view
      class="at-card__content-note"
    >
      小Tips
    </view>
  </view>
</view>
`;

exports[`Card should render prop -- extraStyle 1`] = `
<view
  class="at-card at-card--full"
>
  <view
    class="at-card__header"
  >
    <view
      class="at-card__header-thumb"
    >
      <image
        class="at-card__header-thumb-info"
        mode="scaleToFill"
        src="http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png"
      />
    </view>
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    <text
      class="at-card__header-extra"
      style="font-size: 12px; max-width: 200px;"
    >
      额外信息
    </text>
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    <view
      class="at-card__content-note"
    >
      小Tips
    </view>
  </view>
</view>
`;

exports[`Card should render prop -- isFull 1`] = `
<view
  class="at-card at-card--full"
>
  <view
    class="at-card__header"
  >
    <view
      class="at-card__header-thumb"
    >
      <image
        class="at-card__header-thumb-info"
        mode="scaleToFill"
        src="http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png"
      />
    </view>
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    <text
      class="at-card__header-extra"
    >
      额外信息
    </text>
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    <view
      class="at-card__content-note"
    >
      小Tips
    </view>
  </view>
</view>
`;

exports[`Card should render prop -- note 1`] = `
<view
  class="at-card"
>
  <view
    class="at-card__header"
  >
    <view
      class="at-card__header-thumb"
    >
      <image
        class="at-card__header-thumb-info"
        mode="scaleToFill"
        src="http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png"
      />
    </view>
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    <view
      class="at-card__content-note"
    >
      小Tips
    </view>
  </view>
</view>
`;

exports[`Card should render prop -- thumb 1`] = `
<view
  class="at-card"
>
  <view
    class="at-card__header"
  >
    <view
      class="at-card__header-thumb"
    >
      <image
        class="at-card__header-thumb-info"
        mode="scaleToFill"
        src="http://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png"
      />
    </view>
    <!---->
    <!---->
    <text
      class="at-card__header-title"
    >
      这是个标题
    </text>
    
  </view>
  <view
    class="at-card__content"
  >
    <view
      class="at-card__content-info"
    >
      这也是内容区 可以随意定义功能
    </view>
    
  </view>
</view>
`;
